soa-derive-internal/src/ptr.rs

```diff
@@ -46,8 +66,12 @@ pub fn derive(input: &Input) -> TokenStream {
         #[derive(Copy, Clone)]
         #visibility struct #ptr_name {
             #(
-                #[doc = #fields_doc]
-                pub #fields_names_1: *const #fields_types,
+                #[doc = #unnested_fields_doc]
+                pub #unnested_fields_names: *const #unnested_fields_types,
+            )*
+            #(
+                #[doc = #nested_fields_doc]
+                pub #nested_fields_names: <#nested_fields_types as soa_derive::SoAPtr>::Ptr,
```

Comment:
This might be problematic if the user wants control over the fields layout, since it is no longer following the same order as the initial source code. We could try to define the helper structs in the same order as the input source; or forbid any `#[repr(...)]` attribute on them.

Reply:
Yeah, that's a problem, I will try this:

> define the helper structs in the same order as the input source

Reply:
Fixed, I introduced two methods `Input::fields_seq` and `Input::fold_fields`, which keep the generated tokens as same order as the original struct's fields.